server/routes/offenceCodeDecisions/offenceCodeDecisions.ts

```diff
@@ -4,72 +4,90 @@ import decisionTree from '../../offenceCodeDecisions/DecisionTree'
 import IncidentRole from '../../incidentRole/IncidentRole'
-import { properCaseName } from '../../utils/utils'
+import { properCaseName, formatName } from '../../utils/utils'
+import { DecisionForm } from './decisionForm'
+import {
+  decisionFormFromPost,
+  getAndDeleteSessionDecisionForm,
+  getRedirectUrlForUserSearch,
+  setSessionDecisionForm,
+  updateSessionDecisionForm,
+} from './offenceCodeDecisionsSessionHelper'
+import { DecisionType } from '../../offenceCodeDecisions/Decision'
+import { User } from '../../data/hmppsAuthClient'
 
-type PageData = {
-  error?: FormError
-  selectedDecisionId?: string
-}
+type PageData = { errors?: FormError[] } & DecisionForm
 
 export default class OffenceCodeRoutes {
-  constructor(private readonly placeOnReportService: PlaceOnReportService) {}
-
-  private async placeholderValues(adjudicationNumber: string, res: Response) {
-    const { user } = res.locals
-    const draftAdjudication = await this.placeOnReportService.getDraftAdjudicationDetails(
-      Number(adjudicationNumber),
-      user
-    )
-    const prisonerDetails = await this.placeOnReportService.getPrisonerDetails(
-      draftAdjudication.draftAdjudication.prisonerNumber,
-      user
-    )
-    return {
-      offenderFirstName: properCaseName(prisonerDetails.firstName),
-      offenderLastName: properCaseName(prisonerDetails.lastName),
-    }
-  }
+  constructor(private readonly placeOnReportService: PlaceOnReportService, private readonly userService: UserService) {}
 
   private renderView = async (req: Request, res: Response, pageData?: PageData): Promise<void> => {
     const { adjudicationNumber, incidentRole } = req.params
-    const { error, selectedDecisionId } = pageData
-    const placeholderValues = await this.placeholderValues(adjudicationNumber, res)
-    const path = req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, '')
-    const decision = decisionTree.findByUrl(path)
-
+    const { errors } = pageData
+    const { user } = res.locals
+    const decisionForm = pageData // The form backing this page
+    const placeholderValues = await this.placeholderValues(adjudicationNumber, user)
+    const decision = decisionTree.findByUrl(req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, ''))
     const pageTitle = decision.getTitle().getProcessedText(placeholderValues, incidentRole as IncidentRole)
     const questions = decision.getChildren().map(d => {
       return {
         id: d.id(),
         label: d.getQuestion().getProcessedText(placeholderValues),
+        type: d.getType().toString(),
       }
     })
-    return res.render(`pages/${decision.getPage() || 'offenceCodeDecisions'}`, {
-      errors: error ? [error] : [],
-      selectedDecisionId,
+    const selectedDecisionViewData = await this.viewDataFromDecisionForm(decisionForm, user)
+    return res.render(`pages/offenceCodeDecisions`, {
+      errors: errors || [],
+      decisionForm,
+      selectedDecisionViewData,
```

Comment:
This is extra view data that we need to show depending on what has been chosen in the form backing object - e.g. if they have selected a prisoner - we need the prisoner name